flang/lib/Lower/CallInterface.cpp

```diff
@@ -423,7 +423,7 @@ class Fortran::lower::CallInterfaceImpl {
                     getResultEntity(interface.side().getCallDescription()));
     auto lenTy = mlir::IndexType::get(&mlirContext);
     auto charRefTy = fir::ReferenceType::get(
-        fir::CharacterType::get(&mlirContext, type.kind()));
+        fir::CharacterType::get(&mlirContext, type.kind(), fir::CharacterType::unknownLen()));
```

Comment:
Using `getUnknownLen` would be more the MLIR style.

Reply:
Added `fir::CharacterType::getUnkownLen(ctxt, kind)`.